modules/secrets-store-azure.adoc

```diff
@@ -0,0 +1,313 @@
+
+- Added the ClusterRole and ClusterRoleBinding
+- Added namespace to the serviceaccount and daemonset
+- Changed securityContext to privileged on the container
+
+See if all of these are correct or not. Removed the "oc adm policy add-scc-to-use" step
+////
+
+.. Grant privileged access to the `csi-secrets-store-provider-azure` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z csi-secrets-store-provider-azure -n openshift-cluster-csi-drivers
+----
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f azure-provider.yaml
+----
+
+. TODO what is the summary of what we're doing here?
+
+.. Set the service principal client secret as an environment variable by running the following command:
++
+[source,terminal]
+----
+$ SERVICE_PRINCIPAL_CLIENT_SECRET="$(az ad sp create-for-rbac --name https://$KEYVAULT_NAME --query 'password' -otsv)"
+----
+
+.. Set the service principal client ID as an environment variable by running the following command:
++
+[source,terminal]
+----
+$ SERVICE_PRINCIPAL_CLIENT_ID="$(az ad sp list --display-name https://$KEYVAULT_NAME --query '[0].appId' -otsv)"
+----
+
+.. Create a generic secret witht he service principal client secret and ID by running the following command:
++
+[source,terminal]
+----
+$ oc create secret generic secrets-store-creds -n my-namespace --from-literal clientid=${SERVICE_PRINCIPAL_CLIENT_ID} --from-literal clientsecret=${SERVICE_PRINCIPAL_CLIENT_SECRET}
+----
+
+.. Label the secret with TODO by running the following command:
++
+[source,terminal]
+----
+$ oc -n my-namespace label secret secrets-store-creds secrets-store.csi.k8s.io/used=true
+----
+
+. Create a secret provider class to define your secrets store provider:
+
+.. Create a YAML file that defines the `SecretProviderClass` object:
++
+.Example `secret-provider-class-azure.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-azure-provider                 <1>
+  namespace: my-namespace                 <2>
+spec:
+  provider: azure                         <3>
+  parameters:                             <4>
+    usePodIdentity: "false"
+    useVMManagedIdentity: "false"
+    userAssignedIdentityID: ""
+    keyvaultName: "kvname"
+    objects: |
+      array:
+        - |
+          objectName: secret1
+          objectType: secret
+    tenantId: "tid"
+----
+<1> Specify the name for the secret provider class.
+<2> Specify the namespace for the secret provider class.
+<3> Specify the provider as `azure`.
+<4> Specify the provider-specific configuration parameters.
+
+.. Create the `SecretProviderClass` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret-provider-class-azure.yaml
+----
+
+. Create a deployment to use this secret provider class:
+
+.. Create a YAML file that defines the `Deployment` object:
```

Comment:
As i mentioned in our earlier discussion: https://redhat-internal.slack.com/archives/C05CXR1PVRD/p1694765329809519?thread_ts=1694466510.194569&cid=C05CXR1PVRD 
Better to use pod instead of deployment 

```suggestion
apiVersion: v1
kind: Pod
metadata:
  name: mypod-1secret
  namespace: testropatil
spec:
  containers:
  - image: quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
      runAsUser: 1000
      seccompProfile:
        type: RuntimeDefault
    name: mypod
    volumeMounts:
    - name: secrets-store-inline
      mountPath: "/mnt/secret-store"
      readOnly: true
  volumes:
    - name: secrets-store-inline
      csi:
        driver: secrets-store.csi.k8s.io
        readOnly: true
        volumeAttributes:
          secretProviderClass: "secret-store-wutyenstyk-1secret"
        nodePublishSecretRef:
          name: secrets-store-creds
```

Reply:
Customers are free to use pod, deployment or any other replica controller, all should work. So, I don't know whether you are meaning deployment does not work and only pod works. If so, it should be a bug.